src/python/pants/backend/python/lint/isort/rules.py

```diff
@@ -105,9 +118,41 @@ async def isort_fmt(request: IsortRequest, isort: Isort) -> FmtResult:
     return FmtResult.create(request, result, output_snapshot, strip_chroot_path=True)
 
 
+@rule(desc="Export virtualenv for isort", level=LogLevel.DEBUG)
+async def isort_export(request: IsortExportRequest, isort: Isort, pex_pex: PexPEX) -> ExportResult:
+    isort_pex = await Get(Pex, PexRequest, isort.to_pex_request())
+
+    dest = os.path.join("python", "virtualenvs", isort.name)
+
+    merged_digest = await Get(Digest, MergeDigests([pex_pex.digest, isort_pex.digest]))
+    pex_pex_path = os.path.join("{digest_root}", pex_pex.exe)
+    return ExportResult(
+        f"virtualenv for the tool '{isort.name}'",
+        dest,
+        digest=merged_digest,
+        post_processing_cmds=[
+            PostProcessingCommand(
+                [
+                    pex_pex_path,
+                    os.path.join("{digest_root}", isort_pex.name),
+                    "venv",
+                    "--collisions-ok",
+                    "--remove=all",
+                    # FIXME: If I add an additional directory layer after the digest_root,
+                    # the export works. What value would make sense?
+                    "{digest_root}/",
```

Comment:
Good debugging! Any ideas @benjyw?

In the meantime, to unblock yourself, I recommend that you use a value like `foo`. Figuring out this weird behavior is polish, and it can be done at any time and in parallel to other work.

Reply:
Figured something out: If I drop the `isort.name` from the definition of `dest`, then I can use `"{digest_root}/isort"` here and things work as expected